### ISSUE TITLE
Workflows relying on resource template fail on arm64

### ISSUE THREAD

**Issue by radusora** [argoproj/argo-workflows#10538]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [x] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

After switching to v3.4.5 workflows that rely on the resource template fail when running on arm64. Basically the changes introduced here (https://github.com/argoproj/argo-workflows/commit/6673cfbdfaec5c8082b7c37516024e9fe6c35859) use bitnami/kubectl:1.24.8, which is only available for amd64 and not arm64. This means that the argoexec image is built for amd64 but includes an amd64 binary for kubectl.

### Version

v.3.4.5

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
time="2023-02-15T23:17:45.632Z" level=info msg="kubectl create -f /tmp/manifest.yaml -o json"
time="2023-02-15T23:17:45.633Z" level=warning msg="Non-transient error: fork/exec /bin/kubectl: exec format error"
time="2023-02-15T23:17:45.633Z" level=error msg="executor error: no more retries fork/exec /bin/kubectl: exec format error"
```

### Logs from the workflow controller

```text
Not relevant
```

### Logs from in your workflow's wait container

```text
Not relevant
```

**Comment by JPZ13** [comment]
Thanks for leaving the comment. @radusora - do you have a suggestion for a base kubectl image that supports both arm and amd? If so, could you put out a quick PR with it added instead of the bitnami image?

cc @juliev0

**Comment by radusora** [comment]
@JPZ13 The only thing that comes close to being acceptable is [alpine/k8s](https://hub.docker.com/r/alpine/k8s) as they recently introduced multi arch support. However, it's 3 times larger than the kubectl one being used now. Depending on your motivation to move away from curl, it might or might not make sense to use it. Let me know!

**Comment by JPZ13** [comment]
@radusora - alpine/k8s is still a small image at ~200mb. I think it would be better to use than curling it ourselves. Can you put out a PR for it?

**Comment by radusora** [comment]
@JPZ13 Sounds good, will do!

**Comment by radusora** [comment]
@JPZ13 Realised that this will also be fixed by #10550 which should be better than just swapping the image as discussed
cc @johejo

**Comment by caelan-io** [comment]
Thanks for posting an update @radusora. Closing this since #10550 merged.
